### PR TITLE
feat: Introduce methods to set SPP and fetch OTP from the secondary server

### LINKED ISSUE #1219: Introduce methods to set SPP and fetch OTP in at_client SDK
Introduce two new methods to set Semi Permanent Pass-code (SPP) and fetch the OTP  from the server in at_client SDK

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
- Introduce "setSPP" and "getOTP" methods in at_client_spec to set the semi permanent pass-code and fetch the OTP from the secondary server.
- Introduce "enrollmentId" as optional parameter to AtClient.create method and AtClient._init method to set the enrollment Id which will be propagated to the AtLookup instance used for APKAM authentication.
- Update the at_demo_data from v1.0.1 to v1.0.3 which has RSA Key-pair added APKAM.
- In the "at_keys_initializer.dart" file, eliminate the dependency on "at_demo_credentials.dart," as it duplicates the content found in the "at_demo_data.dart" file.
- Update the test condition in the enrollment_test.dart - "validate client functionality to fetch pending enrollments on legacy pkam authenticated client"
  The initial condition was 
```
    expect(enrollmentRequests.length, 4);
   // 4 entries - 2 entries from this test
   // + 2 entries from the other test in this file.
```
With this condition, there might be intermitent failure depending on the order of the test run. Hence the condition as follows - `expect(enrollmentRequests.length > 2, true);`

**- How I did it**
- The setSPP method accepts a String which represents an SPP. Performs validations and if successful, sends the SPP to secondary server via the executeCommand, else returns an InvalidPINException.
- The getOTP method fetches the OTP from the secondary server and returns to the client.

**- How to verify it**
- Added unit tests to assert on the negative scenarios that returns InvalidPINException for an Invalid SPP supplied.
- Added functional test to create a new SPP and submit and enrollment request with the SPP.

**- Description for the changelog**
- Introduce methods to set SPP and fetch OTP from the secondary server

### CODE REVIEW

**Review by gkc — COMMENTED** [review]
LGTM - I won't approve, I'll let @murali-shris approve once he has reviewed

File: packages/at_client/lib/src/client/at_client_spec.dart
Comment (murali-shris) [inline packages/at_client/lib/src/client/at_client_spec.dart:549]: document the exception when an unauthorized connection calls setSPP

File: tests/at_functional_test/pubspec.yaml
Comment (murali-shris) [inline tests/at_functional_test/pubspec.yaml]: why can't we use published at_demo_data?

**Review by murali-shris — CHANGES_REQUESTED** [review]

File: tests/at_functional_test/pubspec.yaml
Comment (sitaram-kalluri) [reply tests/at_functional_test/pubspec.yaml]: Sure Murali, will publish the at_demo_data and update the version here.

File: packages/at_client/lib/src/client/at_client_impl.dart
Comment (murali-shris) [inline packages/at_client/lib/src/client/at_client_impl.dart]: have this line in AtClientImpl._ instead of _init(). you can remove the param to _init

File: packages/at_client/lib/src/client/at_client_impl.dart
Comment (murali-shris) [inline packages/at_client/lib/src/client/at_client_impl.dart]: we have to check the error code and then throw InvalidPinException. It can be any AtLookUpException here

**Review by murali-shris — CHANGES_REQUESTED** [review]

File: packages/at_client/lib/src/client/at_client_impl.dart
Comment (sitaram-kalluri) [reply packages/at_client/lib/src/client/at_client_impl.dart]: This is addressed in [38d3293](https://github.com/atsign-foundation/at_client_sdk/pull/1223/commits/38d3293920af315d493a991a59b61eb49e201704)

File: packages/at_client/lib/src/client/at_client_impl.dart
Comment (sitaram-kalluri) [reply packages/at_client/lib/src/client/at_client_impl.dart]: Added validations for the SPP before the invocation of remoteSecondary.executeCommand. So for any invalid pin, the exception will be returned from at_client itself (Calls will not be send to secondary server). Hence converting any exception from secondary server as `AtClientException`

**Review by murali-shris — APPROVED** [review]

**Pull request merged**
